TSOClient/tso.simantics/NetPlay/EODs/Handlers/VMEODSlotsPlugin.cs

```diff
@@ -0,0 +1,678 @@
+                    VM.SendCommand(new VMNetAsyncResponseCmd(0, new VMTransferFundsState
+                {
+                    Responded = true,
+                    Success = success,
+                    TransferAmount = transferAmount,
+                    UID1 = uid1,
+                    Budget1 = budget1,
+                    UID2 = uid2,
+                    Budget2 = budget2
+                }));
+                if (success)
+                {
+                    client.Send("slots_resume_manage", amountString);
+                }
+                else
+                {
+                    Console.WriteLine("VMEODSlotsPlugin.WithdrawHandler: There was an error trying to withdraw money to the owner!");
+                    client.Send("slots_resume_manage", "0");
+                }
+            });
+
+            // client.SendOBJEvent(new VMEODEvent((short)VMOEDSlotsObjectEvents.WithdrawCash)); *Depricated*
+        }
+        private void DepositHandler(string evt, string amountString, VMEODClient client)
+        {
+            // parse the withdraw amount
+            short depositAmount = Int16.Parse(amountString);
+
+            // atempt to credit the machine by debiting the owner
+            var VM = client.vm;
+            VM.GlobalLink.PerformTransaction(VM, false, client.Avatar.PersistID, Server.Object.PersistID, Convert.ToInt32(depositAmount),
+
+            (bool success, int transferAmount, uint uid1, uint budget1, uint uid2, uint budget2) =>
+            {
+                //TODO: Make this part of global link
+                VM.SendCommand(new VMNetAsyncResponseCmd(0, new VMTransferFundsState
+                {
+                    Responded = true,
+                    Success = success,
+                    TransferAmount = transferAmount,
+                    UID1 = uid1,
+                    Budget1 = budget1,
+                    UID2 = uid2,
+                    Budget2 = budget2
+                }));
+                if (success)
+                {
+                    client.Send("slots_resume_manage", "-" + amountString);
+                }
+                else
+                {
+                    client.Send("slots_deposit_fail", "" + amountString);
+                }
+            });
+        }
+        private void GameOverHandler(short eventID, VMEODClient player)
+        {
+            // check if object has enough money or too much money, then start new game or disconnect the player
+            if (IsObjectBalanceInBounds() == true)
+            {
+                player.Send("slots_new_game", "");
+                State = VMEODSlotsStates.Idle;
+            }
+            else
+            {
+                State = VMEODSlotsStates.Off;
+                userClient.Send("slots_toggle_offline_message", "");
+            }
+        }
+        private void BetHandler(string evt, string betAmountString, VMEODClient client)
+        {
+            short betAmount = Int16.Parse(betAmountString);
```

Comment:
The user can literally steal money from the machine if they give a bad value here, and you haven't bounded what they can bet. If a user submitted a request with a modified client to bet a million simoleons, it would do it.